client/my-sites/plan-features/test/index.jsx

```diff
@@ -95,49 +116,44 @@ describe( 'isPrimaryUpgradeByPlanDelta', () => {
 
 describe( 'PlanFeatures.renderUpgradeDisabledNotice', () => {
 	const baseProps = {
+		planProperties: [],
+		recordTracksEvent: jest.fn(),
 		translate: identity,
 	};
 
 	const originalCreatePortal = ReactDOM.createPortal;
 	beforeAll( () => {
 		ReactDOM.createPortal = ( elem ) => elem;
+		jest.spyOn( document, 'querySelector' ).mockImplementation( ( selector ) => {
```

Comment:
could we add a comment about why we need this? I still don't fully understand what it's accomplishing after reading the PR 😅

Reply:
Sure thing! Added a comment in 21d44f19f22b80ef91b29a305d7729aeed7ff59a. 

Basically, the original component doesn't render a `.plans-features-main__notice` so we render it ourselves. However, JSDOM's can't find it that way, because it's outside of the component we're rendering, so we need this in order to trick the `getBannerContainer()` method to find the container here:

https://github.com/Automattic/wp-calypso/blob/24a8c5cfafd89c2ebabb71551cc2dcde60be2a73/client/my-sites/plan-features/index.jsx#L267

And this container is necessary for the component logic to work in a few areas.

Let me know if that makes sense.